stdlib/public/Darwin/Foundation/Data.swift

```diff
@@ -2436,15 +2466,26 @@ public struct Data : ReferenceConvertible, Equatable, Hashable, RandomAccessColl
     /// - parameter newElements: The replacement bytes.
     @inlinable // This is @inlinable as generic and reasonably small.
     public mutating func replaceSubrange<ByteCollection : Collection>(_ subrange: Range<Index>, with newElements: ByteCollection) where ByteCollection.Iterator.Element == Data.Iterator.Element {
+        // If the collection is already contiguous, access the underlying raw memory directly.
+        if let contiguous = newElements as? ContiguousBytes {
+            contiguous.withUnsafeBytes { buffer in
+                _representation.replaceSubrange(subrange, with: buffer.baseAddress, count: buffer.count)
+            }
+            return
+        }
+        // The collection might still be able to provide direct access to typed memory.
+        // NOTE: It's safe to do this because we're already guarding on ByteCollection's element as `UInt8`. This would not be safe on arbitrary collections.
+        let replaced: Void? = newElements.withContiguousStorageIfAvailable { buffer in
```

Comment:
`Void?` as a boolean? that seems almost UB territory no?

Reply:
It's a bit unusual, but the behavior is well-defined. `Void` is very much distinct from `nil`, so there is zero chance of a type system confusion.

I initially had this return a dummy boolean, but I felt it added irrelevant detail that obscured the intent of the code:

```swift
        let replaced: Bool? = newElements.withContiguousStorageIfAvailable { buffer in
             _representation.replaceSubrange(subrange, with: buffer.baseAddress, count: buffer.count)
             return true
         }
         guard replaced == nil else { return }
```

The `withContiguousStorageIfAvailable` API kind of forces this sort of weird control flow. If we were to design it now, I'd argue it should return a `Result`-like enum rather than an `Optional` -- but that's too late now.